scope_registry.go

```diff
@@ -88,28 +113,33 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 	}
 
 	buf := keyForPrefixedStringMapsAsKey(make([]byte, 0, 256), prefix, parent.tags, tags)
-	r.mu.RLock()
+	h := maphash.Hash{}
+	h.SetSeed(r.seed)
+	_, _ = h.Write(buf)
+	ptrToSubscope := r.subscopes[h.Sum64()%uint64(len(r.subscopes))]
```

Comment:
I did this because there was a comment before from vytenis saying that I was actually getting the copy of a lock. So I wanted the name to show that this is a pointer.

Reply:
Eh sure, usually that's just enforced via the types but not too terrible as is.

Reply:
I got confused initially, but thinking about it more, go vet would most certainly catch this.

Reply:
feel free to revert to previous naming :)